src/main/java/uk/gov/justice/digital/hmpps/keyworker/controllers/KeyworkerServiceController.kt

```diff
@@ -0,0 +1,291 @@
+
+    @ApiOperation(value = "Deallocate a key worker from an offender", notes = "Marks the offender with expired time on active record", authorizations = [Authorization("OMIC_ADMIN")], nickname = "deallocate")
+    @ApiResponses(value = [ApiResponse(code = 200, message = "De allocated"), ApiResponse(code = 400, message = "Invalid request.", response = ErrorResponse::class), ApiResponse(code = 404, message = "Requested resource not found.", response = ErrorResponse::class), ApiResponse(code = 500, message = "Unrecoverable error occurred whilst processing request.", response = ErrorResponse::class)])
+    @PutMapping(path = ["/deallocate/{offenderNo}"])
+    fun deallocate(
+        @ApiParam(value = "offenderNo", required = true) @PathVariable("offenderNo") offenderNo: String
+    ) = keyworkerService.deallocate(offenderNo)
+
+    @ApiOperation(value = "Process manual allocation of an offender to a Key worker.", notes = "Process manual allocation of an offender to a Key worker.", authorizations = [Authorization("OMIC_ADMIN")], nickname = "allocate")
+    @ApiResponses(value = [ApiResponse(code = 201, message = "The allocation has been created.")])
+    @PostMapping(path = ["/allocate"], consumes = [MediaType.APPLICATION_JSON_VALUE])
+    fun allocate(
+        @ApiParam(value = "New allocation details.", required = true) @RequestBody keyworkerAllocation: @Valid KeyworkerAllocationDto
+    ): ResponseEntity<*> {
+        keyworkerService.allocate(keyworkerAllocation)
+        return ResponseEntity.status(HttpStatus.CREATED).build<Any>()
+    }
+
+    @ApiOperation(value = "Search for key workers within prison.", notes = "Search for key workers using firstname or lastname", nickname = "keyworkersearch")
+    @ApiResponses(value = [ApiResponse(code = 200, message = "OK", response = KeyworkerDto::class, responseContainer = "List"), ApiResponse(code = 400, message = "Invalid request.", response = ErrorResponse::class), ApiResponse(code = 404, message = "Requested resource not found.", response = ErrorResponse::class), ApiResponse(code = 500, message = "Unrecoverable error occurred whilst processing request.", response = ErrorResponse::class)])
+    @GetMapping(path = ["/{prisonId}/members"])
+    fun keyworkerSearch(
+        @ApiParam(value = "prisonId", required = true) @PathVariable("prisonId") prisonId: String,
+        @ApiParam(value = "Filter results by first name and/or last name of key worker. Supplied filter term is matched to start of key worker's first and last name.") @RequestParam(value = "nameFilter") nameFilter: Optional<String>?,
+        @ApiParam(value = "Filter results by status of key worker.") @RequestParam(value = "statusFilter") statusFilter: Optional<KeyworkerStatus>?,
+        @ApiParam(value = "Requested offset of first record in returned collection of allocation records.", defaultValue = "0") @RequestHeader(value = PagingAndSortingDto.HEADER_PAGE_OFFSET, defaultValue = "0") pageOffset: Long?,
```

Comment:
you've set a default value here but still have the parameter as a `Long?`

Reply:
I'll fix for all the methods